### ISSUE TITLE
QT threads error on 3d_ldm_tutorial.ipynb

### ISSUE THREAD

**Issue by IanHOBD** [Project-MONAI/tutorials#1894]
I have copied the python code from

3d_ldm_tutorial.ipynb

into a python file and I am running it using python3.10 under Ubuntu 22.

The call 

fig, axs = plt.subplots(nrows=1, ncols=3)

in this section:

`check_data = first(train_loader)
idx = 0

img = check_data["image"][idx, 0]
fig, axs = plt.subplots(nrows=1, ncols=3)
for ax in axs:
ax.axis("off")
ax = axs[0]
ax.imshow(img[..., img.shape[2] // 2], cmap="gray")
ax = axs[1]
ax.imshow(img[:, img.shape[1] // 2, ...], cmap="gray")
ax = axs[2]
ax.imshow(img[img.shape[0] // 2, ...], cmap="gray")`

causes the following problem with qt:

QObject::moveToThread: Current thread (0x596fff9c53d0) is not the object's thread (0x596fffa53cd0).
Cannot move to target thread (0x596fff9c53d0)

Any idea why? It doesn't look like it should be running on a different thread. Only thing I can think of is that the data loader worker gets check_data owned by a different thread, but I presume this code worked for the author.

**Comment by ericspod** [comment]
Looking at other cases where this exception comes up, it may be related to conflicting Qt installs. Are you using Anaconda? Can you give an export of the environment you're using? If you're mixing installing pyqt, OpenCV, and Matplotlib between conda and pip this can possibly cause conflicts, everything should be installed with one or the other.

**Comment by IanHOBD** [comment]
Hi Eric.  I'm not using anaconda, just Pip.  Here's my pip freeze:

absl-py==2.1.0
aiohappyeyeballs==2.4.0
aiohttp==3.10.5
aiosignal==1.3.1
alembic==1.14.0
amqp==5.2.0
appdirs==1.4.4
apsw==3.36.0.post1
apturl==0.5.2
argcomplete==3.4.0
asciitree==0.3.3
asgiref==3.7.2
astor==0.8.1
async-timeout==4.0.3
attrs==24.2.0
awscli==1.22.22
backcall==0.2.0
bcrypt==3.2.0
beautifulsoup4==4.10.0
beniget==0.4.1
billiard==4.2.0
blinker==1.9.0
boto3==1.20.22
botocore==1.23.22
Brlapi==0.8.3
Brotli==1.0.9
cached-property==1.5.2
cachetools==5.5.0
celery==5.3.6
certifi==2024.2.2
cffi==1.16.0
chardet==5.2.0
charset-normalizer==2.0.12
clearml==1.17.0rc0
click==8.1.7
click-didyoumean==0.3.0
click-plugins==1.1.1
click-repl==0.3.0
cloudpickle==3.1.0
colorama==0.4.3
coloredlogs==15.0.1
colorlog==6.9.0
command-not-found==0.3
ConfigArgParse==1.5.3
contourpy==1.3.1
cryptography==42.0.4
css-parser==1.0.7
cssselect==1.1.0
cssutils==1.0.2
cucim-cu12==24.10.0
cupshelpers==1.0
cupy-cuda12x==13.3.0
cycler==0.12.1
databricks-sdk==0.38.0
dataclasses==0.6
dbus-python==1.2.18
decorator==4.3.0
defer==1.0.6
Deprecated==1.2.15
dicom2nifti==2.4.11
distro==1.7.0
distro-info==1.1+ubuntu0.2
Django==4.2.3
django-celery-results==2.4.0
dnspython==2.1.0
docker==7.1.0
docutils==0.15.2
duplicity==0.8.21
einops==0.8.0
entrypoints==0.3
fasteners==0.14.1
fastrlock==0.8.2
feedparser==6.0.8
filelock==3.11.0
fire==0.7.0
Flask==3.1.0
flatbuffers==24.3.25
fonttools==4.43.0
frozenlist==1.4.1
fs==2.4.12
fsspec==2024.10.0
furl==2.1.3
future==0.18.3
gast==0.5.2
gdown==5.2.0
gitdb==4.0.11
GitPython==3.1.43
google-auth==2.36.0
graphene==3.4.3
graphql-core==3.2.5
graphql-relay==3.2.0
greenlet==3.1.1
grpcio==1.68.1
gunicorn==23.0.0
h5py==3.10.0
html2text==2020.1.16
html5-parser==0.4.10
html5lib==1.1
httplib2==0.20.2
huggingface-hub==0.26.5
humanfriendly==10.0
humanize==4.11.0
idna==3.7
ifaddr==0.1.7
imagecodecs==2024.9.22
imageio==2.13.3
importlib_metadata==8.5.0
importlib_resources==6.4.5
inquirerpy==0.3.4
ipython==7.31.1
itk==5.4.0
itk-core==5.4.0
itk-filtering==5.4.0
itk-io==5.4.0
itk-numerics==5.4.0
itk-registration==5.4.0
itk-segmentation==5.4.0
itsdangerous==2.2.0
jedi==0.18.0
jeepney==0.7.1
Jinja2==3.1.4
jmespath==0.10.0
joblib==1.3.2
json-tricks==3.17.3
jsonschema==4.23.0
jsonschema-specifications==2024.10.1
keyring==23.5.0
kiwisolver==1.4.5
kombu==5.3.5
language-selector==0.1
launchpadlib==1.10.16
lazr.restfulclient==0.14.4
lazr.uri==1.0.6
lazy_loader==0.4
lmdb==1.5.1
lockfile==0.12.2
louis==3.20.0
lpips==0.1.4
lxml==4.8.0
lz4==3.1.3+dfsg
macaroonbakery==1.3.1
Mako==1.1.3
Markdown==3.3.6
markdown-it-py==3.0.0
MarkupSafe==3.0.2
matplotlib==3.7.3
matplotlib-inline==0.1.3
mdurl==0.1.2
mechanize==0.4.7
MedPy==0.4.0
mlflow==2.18.0
mlflow-skinny==2.18.0
monai==1.4.0
monai-generative==0.2.3
monotonic==1.6
more-itertools==8.10.0
mpmath==1.3.0
msgpack==1.0.3
multidict==6.0.5
mysqlclient==2.1.1
natsort==8.4.0
ndindex==1.5.2
nest-asyncio==1.6.0
netifaces==0.11.0
networkx==3.2.1
neurocombat-sklearn==0.1.3
nibabel==5.2.1
ninja==1.11.1.2
nni==3.0
numcodecs==0.13.1
numpy==1.26.4
nvidia-cublas-cu11==11.10.3.66
nvidia-cublas-cu12==12.4.5.8
nvidia-cuda-cupti-cu12==12.4.127
nvidia-cuda-nvrtc-cu11==11.7.99
nvidia-cuda-nvrtc-cu12==12.4.127
nvidia-cuda-runtime-cu11==11.7.99
nvidia-cuda-runtime-cu12==12.4.127
nvidia-cudnn-cu11==8.5.0.96
nvidia-cudnn-cu12==9.1.0.70
nvidia-cufft-cu12==11.2.1.3
nvidia-curand-cu12==10.3.5.147
nvidia-cusolver-cu12==11.6.1.9
nvidia-cusparse-cu12==12.3.1.170
nvidia-ml-py==12.560.30
nvidia-nccl-cu12==2.21.5
nvidia-nvjitlink-cu12==12.4.127
nvidia-nvtx-cu12==12.4.127
oauthlib==3.2.0
olefile==0.46
onnx==1.17.0
onnxruntime==1.20.1
opencv-python==4.10.0.84
openslide-python==1.4.1
opentelemetry-api==1.28.2
opentelemetry-sdk==1.28.2
opentelemetry-semantic-conventions==0.49b2
optuna==4.1.0
ordered-set==4.0.2
orderedmultidict==1.0.1
packaging==21.3
pandas==1.3.4
paramiko==2.9.3
parso==0.8.1
pathlib2==2.3.7.post1
patsy==0.5.2
pexpect==4.8.0
pfzy==0.3.4
pickleshare==0.7.5
pillow==10.3.0
plotly==5.4.0
ply==3.11
prettytable==3.12.0
prompt-toolkit==3.0.43
protobuf==5.29.1
psutil==5.9.8
ptyprocess==0.7.0
py7zr==0.11.3+dfsg
pyamg==5.2.1
pyarrow==18.1.0
pyasn1==0.4.8
pyasn1_modules==0.4.1
pybind11==2.8.1
pycairo==1.20.1
pychm==0.8.6
pycparser==2.21
pycryptodomex==3.11.0
pycups==2.0.1
pydicom==2.2.2
Pygments==2.18.0
PyGObject==3.42.1
PyJWT==2.9.0
pymacaroons==0.13.0
pyminc==0.54
PyNaCl==1.5.0
pynetdicom==1.5.7
pynrrd==1.1.1
pyparsing==3.0.6
PyQt5==5.15.6
PyQt5-sip==12.9.1
PyQtWebEngine==5.15.5
pyRFC3339==1.1
Pyro4==4.81
PySocks==1.7.1
python-apt==2.4.0+ubuntu4
python-dateutil==2.8.2
python-debian==0.1.35
python-gdcm==3.0.24.1
PythonWebHDFS==0.2.3
pythran==0.10.0
pytorch-ignite==0.4.11
pytz==2021.3
PyWavelets==1.2.0
pyxdg==0.26
PyYAML==5.4.1
qbatch==2.3
redbrick-sdk==2.18.1
referencing==0.35.1
regex==2021.11.10
reportlab==3.6.13
repoze.lru==0.7
requests==2.32.0
requests-toolbelt==0.9.1
responses==0.25.3
retrying==1.3.3
rich==13.8.0
roman==3.3
Routes==2.5.1
rpds-py==0.22.3
rsa==4.7.2
rt-utils==1.2.7
s3transfer==0.5.0
safetensors==0.4.5
schema==0.7.7
scikit-image==0.19.0
scikit-learn==1.2.1
scipy==1.14.1
screen-resolution-extra==0.0.0
seaborn==0.11.2
SecretStorage==3.3.1
serpent==1.41
setuptools-scm==6.3.2
sgmllib3k==1.0.0
shellingham==1.5.4
shtab==1.7.1
SimpleITK==2.2.1
simplejson==3.19.3
six==1.17.0
smart-open==5.2.1
smmap==5.0.1
soupsieve==2.3.1
SQLAlchemy==2.0.36
sqlparse==0.5.0
ssh-import-id==5.11
statsmodels==0.13.1
sympy==1.13.1
systemd-python==234
tenacity==8.2.3
tensorboard==2.18.0
tensorboard-data-server==0.7.2
tensorboardX==2.6.2.2
termcolor==2.5.0
texttable==1.6.4
threadpoolctl==3.2.0
tifffile==2024.1.30
tokenizers==0.19.1
tomli==2.0.1
torch==2.5.1
torchio==0.20.3
torchvision==0.20.1
tqdm==4.66.3
traitlets==5.1.1
transformers==4.40.2
triton==3.1.0
typeguard==4.1.2
typer==0.15.1
typing==3.7.4.3
typing_extensions==4.12.2
tzdata==2023.4
ubuntu-drivers-common==0.0.0
ubuntu-pro-client==8001
ufoLib2==0.13.1
ufw==0.36.1
unattended-upgrades==0.1
unicodedata2==14.0.0
urllib3==1.26.18
usb-creator==0.3.7
vine==5.1.0
wadllib==1.3.6
wcwidth==0.2.13
webcolors==1.11.1
webencodings==0.5.1
WebOb==1.8.6
websockets==14.1
Werkzeug==3.1.3
wrapt==1.17.0
xdg==5
xkit==0.0.0
yarl==1.9.4
zarr==2.18.3
zeroconf==0.38.3
zipp==3.21.0

**Comment by ericspod** [comment]
I see a lot of packages in this list so I imagine there's a conflict somewhere with PyQt which is causing this. Is this using the system Python? I wouldn't recommend using that for anything since it's needed by the OS. I would suggest making a virtual environment, or using Anaconda/Miniconda, to create a new environment with the minimal libraries needed for this example, and see if the problem recurs. There's just too many things here to try and sort through what introduces the issue.

**Comment by IanHOBD** [comment]
This worked, thank you.
